docs/scalar_tensor.rst

```diff
@@ -72,7 +72,82 @@ Declaration
         # ERROR: cannot create tensor after any tensor accesses from the Python-scope!
 
 
-Attribute
+Accessing components
+--------------------
+
+.. attribute:: a[p, q, ...]
+
+    :parameter a: (Tensor) the tensor of scalar
+    :parameter p: (scalar) index of the first tensor dimension
+    :parameter q: (scalar) index of the second tensor dimension
+    :return: (scalar) the element at ``[p, q, ...]``
+
+    This extracts the element value at ``[3, 4]`` of tensor ``a``:
+    ::
+
+        x = a[3, 4]
+
+    This sets the element value at ``[3, 4]`` of tensor ``a`` to ``2``:
+    ::
+
+        a[3, 4] = 2
+
+    .. note ::
+
+        The returned value can also be ``Vector`` / ``Matrix`` if ``a`` is a tensor of vector / matrix, see :ref:`vector` for more details.
+
+
+Meta data
 ---------
 
-TODO: WIP
+.. function:: a.dim()
+
+    :parameter a: (Tensor) the tensor
+    :return: (scalar) the length of ``a``
+
+    ::
+
+        x = ti.var(ti.i32, (6, 5))
+        x.dim()  # 2
+
+        y = ti.var(ti.i32, 6)
+        y.dim()  # 1
+
+        z = ti.var(ti.i32, ())
+        z.dim()  # 0
+
+
+.. function:: a.shape()
```

Comment:
This one is actually called get_shape() in python scope.

Reply:
`get_shape()` is a method of `SNode`, in `Expr` it's `shape()`...please check L299 of `expr.py` to verify. We may want to refactor to make `SNode` and `Expr` have the same API later in a separate PR.